rfd/0159-webasset-cache.md

```diff
@@ -34,75 +33,45 @@ web client receives an index.html that will request a file such as
 `index-123123.js` and then later on during the session requests
 `index-234234.js`, if the proxy that is hit when load balancing is of a
 different version than the original that served the `index.html`, that proxy
-will not have the file and return a `404`. Storing all the webassets in a cache
-on auth will allow the proxies to fallback to auth as a single source of truth
-and serve any file needed.
+will not have the file and return a `404`. Storing all the webassets in an s3-compatible
+bucket allows the proxies to fallback to single repository of every webasset in use.
 
 ## Details
 
 ### Webasset cache
 
-The webasset cache is a simple `map[string][]byte` that lives in the auth server.
+When the proxy comes online, we can start an (optional) service that will sync the webassets
+to a configured, s3-compatible storage. When a proxy comes online, it will start a heartbeat
+that first, checks if it's webassets have been loaded. It does this by first, listing the objects in
+the bucket using `ListObjectsV2`. As of right now, we have ~170 items in our webassets bundle. 
+`ListObjectsV2` will only list up to 1,000 items per page. This means that after 5 stored versions
```

Comment:
Too bad you can't list objects by tag - it would make this easy to mitigate.

Reply:
You _can_ list by version. I even did this yesterday during testing. No idea why it left my mind today. Will update, thanks for the reminder